CMakeLists.txt

```
@@ -165,7 +170,12 @@ endif( ${CMAKE_BUILD_TYPE} MATCHES "Release" )
 
 # Set compiler flags
 if(${CMAKE_COMPILER_IS_GNUCXX})
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -funroll-loops -Wall -std=c++11 -Wl,--no-as-needed -ldl")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -funroll-loops -Wall -Wl,--no-as-needed -ldl")
+  if(KRATOS_USE_CXX_17 MATCHES ON)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
```

```suggestion
    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
    
    # Check for standard to use
CHECK_CXX_COMPILER_FLAG(-std=c++17 HAVE_FLAG_STD_CXX17)
IF(HAVE_FLAG_STD_CXX17)
    # Have -std=c++17, use it
    MESSAGE(STATUS "HAVE_FLAG_STD_CXX17 available")
ELSE(HAVE_FLAG_STD_CXX17)
    CHECK_CXX_COMPILER_FLAG(-std=c++1z HAVE_FLAG_STD_CXX1Z)
    IF(HAVE_FLAG_STD_CXX1Z)
        # Have -std=c++1z, use it
        MESSAGE(STATUS "HAVE_FLAG_STD_CXX1Z available")
    ELSE(HAVE_FLAG_STD_CXX1Z)
        # And so on and on...
        MESSAGE(FATAL_ERROR "No modern C++17 found, compilation may fail")
    ENDIF(HAVE_FLAG_STD_CXX1Z)
ENDIF(HAVE_FLAG_STD_CXX17)
```

With this check you can verify the compiler actually supports the standard

---

this is what I would like to add in followup PRs

this PR is simply about adding the flag, easing and improving the implementation should be done in later PRs to not clutter this one

but thx for the suggestion